### ISSUE TITLE
Prevent connection collisions

### ISSUE THREAD

**Issue by BellangerK** [jsplumb/jsplumb#954]
This time I don't come for an issue but a question. Tell me if it's not the place to do so. I'm struggling for days with connection collisions and I was wondering if there was already something in the library to prevent them.

For example, I have this situation where the user connect these 2 elements : 
![image](https://user-images.githubusercontent.com/32840014/87043869-077d7700-c1f6-11ea-81be-815d95b814e1.png)

And then he wants to connect the 2 other elements, this makes the connections unreadable.
![image](https://user-images.githubusercontent.com/32840014/87044224-8672af80-c1f6-11ea-8c85-3302fe128811.png)

My elements are only draggable on Y axis and the default connector stub is 35 (set when i create an endpoint with property `connector: ['Flowchart', { stub: 35}]`. I can handle it by setting again connector stub dynamically on connection event. I do it this way : 
``` typescript
 this.jsPlumbInstance.select({ source: mysource, scope: myscope }).setConnector(["Flowchart", { stub: foundElement + 10 }]);
 this.jsPlumbInstance.select({ source: mysource, scope: myscope }).get(0).addOverlay(["PlainArrow", { location: 1, width: 10, length: 10, id: "arrow" } as ArrowOverlayOptions]);
```
_(Note that i have to addOverlay again because it disappear in the process but probably cause I'm overiding connector)_

So I add 10 to the stub from the previous element. It works well : 
![image](https://user-images.githubusercontent.com/32840014/87045804-9a1f1580-c1f8-11ea-9e9a-82d8adacc88b.png)

but I will have to do lots of coding to handle every situations.
In short, any idea on how to manage collisions would be really helpful. Though if it's out of jsPlumb perimeter I understand, and will warm up for the rest of coding :D
Thanks for your time.

**Comment by sporritt** [comment]
This really comes down to the way the connector works - a fix for this would involve a new connector type, or some major updates to the current connector to make it aware of other connections.  I'll mark this as an enhancement, but I'm not sure if support for this will get into the Community edition or not.  It seems likely the Toolkit edition will support something like this though.

**Comment by KirkRehalOak9** [comment]
Is there any update on this enhancement? This is nearly the exact use-case I am looking to support as well
